Makefile

```diff
@@ -1,123 +1,133 @@
-.PHONY: dep docker pilosa release-build prerelease-build release prerelease prerelease-upload install generate generate-statik generate-protoc statik test cover cover-pkg cover-viz clean docker-build docker-test
+.PHONY: build check-clean clean cover cover-viz default docker docker-build docker-test generate generate-protoc generate-statik install install-build-deps install-dep install-protoc install-protoc-gen-gofast install-statik prerelease prerelease-build prerelease-upload release release-build require-dep require-protoc require-protoc-gen-gofast require-statik test
 
-DEP := $(shell command -v dep 2>/dev/null)
-STATIK := $(shell command -v statik 2>/dev/null)
-PROTOC := $(shell command -v protoc 2>/dev/null)
-VERSION := $(shell git describe --tags 2> /dev/null || echo unknown)
-STATUS := $(shell git status --porcelain)
-IDENTIFIER := $(VERSION)-$(GOOS)-$(GOARCH)
 CLONE_URL=github.com/pilosa/pilosa
-PKGS := $(shell cd $(GOPATH)/src/$(CLONE_URL); go list ./... | grep -v vendor)
-BUILD_TIME=`date -u +%FT%T%z`
+VERSION := $(shell git describe --tags 2> /dev/null || echo unknown)
+VERSION_ID := $(VERSION)-$(GOOS)-$(GOARCH)
+BRANCH := $(if $(TRAVIS_BRANCH),$(TRAVIS_BRANCH),$(shell git rev-parse --abbrev-ref HEAD))
+BRANCH_ID := $(BRANCH)-$(GOOS)-$(GOARCH)
+BUILD_TIME := $(shell date -u +%FT%T%z)
 LDFLAGS="-X github.com/pilosa/pilosa.Version=$(VERSION) -X github.com/pilosa/pilosa.BuildTime=$(BUILD_TIME)"
-DOCKER_GOLANG_IMAGE=golang:latest
-GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
-BRANCH := $(if $(TRAVIS_BRANCH),$(TRAVIS_BRANCH),$(GIT_BRANCH))
-BRANCH_IDENTIFIER := $(BRANCH)-$(GOOS)-$(GOARCH)
+GO_VERSION=latest
 
-default: test pilosa
+# Run tests and compile Pilosa
+default: test build
 
+# Remove vendor and build directories
 clean:
 	rm -rf vendor build
 
-$(GOPATH)/bin:
-	mkdir $(GOPATH)/bin
-
-dep: $(GOPATH)/bin
-	go get -u github.com/golang/dep/cmd/dep
-
+# Set up vendor directory using `dep`
 vendor: Gopkg.toml
-ifndef DEP
-	make dep
-endif
+	$(MAKE) require-dep
```

Comment:
I don't understand how $(MAKE) works - where is that set?

Reply:
The `make` manual advises to always use `$(MAKE)`, which subtly changes a few things. Details are here: https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html